flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/filesystem/FileSystemStateStorageHelper.java

```diff
@@ -57,21 +58,20 @@ public FileSystemStateStorageHelper(Path rootPath, String prefix) throws IOExcep
 
 	@Override
 	public RetrievableStateHandle<T> store(T state) throws Exception {
-		Exception latestException = null;
+		Exception exception = null;
 
 		for (int attempt = 0; attempt < 10; attempt++) {
 			Path filePath = getNewFilePath();
 
 			try (FSDataOutputStream outStream = fs.create(filePath, FileSystem.WriteMode.NO_OVERWRITE)) {
 				InstantiationUtil.serializeObject(outStream, state);
-				return new RetrievableStreamStateHandle<T>(filePath, outStream.getPos());
-			}
-			catch (Exception e) {
-				latestException = e;
+				return new RetrievableStreamStateHandle<>(filePath, outStream.getPos());
+			} catch (Exception e) {
+				exception = ExceptionUtils.firstOrSuppressed(e, exception);
```

Comment:
The exception for each attempt is probably same, so it seems not very necessary for suppressing all the exceptions.  The previous way for keeping last exception should make sense for this case.

`ExceptionUtils.firstOrSuppressed` is more suitable for different exceptions I think.